subsys/bluetooth/controller/hci/hci.c

```diff
@@ -159,19 +172,111 @@ static void reset(struct net_buf *buf, struct net_buf **evt)
 {
 	struct bt_hci_evt_cc_status *ccst;
 
-	ll_reset();
-
 #if CONFIG_BLUETOOTH_CONTROLLER_DUP_FILTER_LEN > 0
 	dup_count = -1;
 #endif
 	/* reset event masks */
 	event_mask = DEFAULT_EVENT_MASK;
 	le_event_mask = DEFAULT_LE_EVENT_MASK;
 
+	if (buf) {
+		ll_reset();
+		ccst = cmd_complete(evt, sizeof(*ccst));
+		ccst->status = 0x00;
+	}
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_TO_HOST_FC)
+	hci_hbuf_total = 0;
+	hci_hbuf_sent = 0;
+	hci_hbuf_acked = 0;
+	conn_count = 0;
+	atomic_set_bit(&hci_state_mask, HCI_STATE_BIT_RESET);
+	k_poll_signal(hbuf_signal, 0x0);
+#endif
+}
+
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_TO_HOST_FC)
+static void set_ctl_to_host_flow(struct net_buf *buf, struct net_buf **evt)
+{
+	struct bt_hci_cp_set_ctl_to_host_flow *cmd = (void *)buf->data;
+	struct bt_hci_evt_cc_status *ccst;
+
 	ccst = cmd_complete(evt, sizeof(*ccst));
-	ccst->status = 0x00;
+
+	/* require host buffer size before enabling flow control, and
+	 * disallow if any connections are up
+	 */
+	if (!hci_hbuf_total || conn_count) {
+		ccst->status = BT_HCI_ERR_CMD_DISALLOWED;
+		return;
+	} else {
+		ccst->status = 0x00;
+	}
+
+	switch (cmd->flow_enable) {
+	case BT_HCI_CTL_TO_HOST_FLOW_DISABLE:
+	case BT_HCI_CTL_TO_HOST_FLOW_ENABLE:
+		hci_hbuf_sent = 0;
+		hci_hbuf_acked = 0;
+		hci_hbuf_total = -hci_hbuf_total;
```

Comment:
There's tons of missing checks like those all over HCI, I have a Jira task to add parameter checking, which would include things like this